services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/migration/DatasetBackFillFileSummary.java

```diff
@@ -0,0 +1,222 @@
+import org.sagebionetworks.repo.model.dbo.dao.NodeUtils;
+import org.sagebionetworks.repo.model.jdo.AnnotationUtils;
+import org.sagebionetworks.repo.model.migration.DatasetBackfillResponse;
+import org.sagebionetworks.util.PaginationIterator;
+import org.sagebionetworks.util.TemporaryCode;
+import org.sagebionetworks.util.ValidateArgument;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallback;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_NODE_ETAG;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_NODE_ID;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_NODE_TYPE;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_REVISION_ENTITY_PROPERTY_ANNOTATIONS_BLOB;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_REVISION_ITEMS;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_REVISION_NUMBER;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_REVISION_OWNER_NODE;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.TABLE_NODE;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.TABLE_REVISION;
+
+@TemporaryCode(author = "sandhra.sokhal@sagebase.org", comment = "This allows back filling of field checksum,size and count for dataset")
+@Service
+public class DatasetBackFillFileSummary {
+    private static final long PAGE_SIZE_LIMIT = 1000;
+    private static final String CHECKSUM = "checksum";
+    private static final String SIZE = "size";
+    private static final String COUNT = "count";
+    private static final String SQL_UPDATE_ETAG = "UPDATE " + TABLE_NODE + " SET " + COL_NODE_ETAG
+            + " = ? WHERE " + COL_NODE_ID + " = ?";
+    private static final String SQL_UPDATE_ANNOTATION = "UPDATE " + TABLE_REVISION + " SET " + COL_REVISION_ENTITY_PROPERTY_ANNOTATIONS_BLOB
+            + " = ? WHERE " + COL_REVISION_OWNER_NODE + " = ? AND " + COL_REVISION_NUMBER + " =?";
+    private static final String SELECT_ALL_DATASET = "SELECT N.ID, R.NUMBER, R.ENTITY_PROPERTY_ANNOTATIONS, R.ITEMS FROM " + TABLE_NODE + " N JOIN " + TABLE_REVISION + " R ON N."
+            + COL_NODE_ID + " = R." + COL_REVISION_OWNER_NODE + " WHERE N." + COL_NODE_TYPE + " ='dataset' ORDER BY N.ID, R.NUMBER LIMIT ? OFFSET ?";
+    private static final RowMapper<DatasetBackFillDTO> DATASET_BACK_FILL_ROW_MAPPER = (rs, rowNum) -> {
```

Comment:
Validataion:

1. write a sql script that can identify the count of the number of datasets that need to be updated.
2. run the admin api call to make the changes
3. validate that zero datasets need to be updated (script from one)
4. Spot check a known datasets/dataset collections that needs backfill and confirm that it worked as expected.